src/test/java/cloud/fogbow/common/plugins/cloudidp/aws/v2/AwsV2IdentityProviderPluginTest.java

```diff
@@ -0,0 +1,150 @@
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.iam.IamClient;
+import software.amazon.awssdk.services.iam.IamClientBuilder;
+import software.amazon.awssdk.services.iam.model.GetUserResponse;
+import software.amazon.awssdk.services.iam.model.User;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ AwsBasicCredentials.class, IamClient.class, StaticCredentialsProvider.class })
+public class AwsV2IdentityProviderPluginTest {
+
+	private static final String ANY_VALUE = "anything";
+	private static final String FAKE_USER_ID = "fake-user-id";
+	
+	private AwsV2IdentityProviderPlugin plugin;
+	
+	@Before
+	public void setUp() {
+		this.plugin = new AwsV2IdentityProviderPlugin();
+	}
+	
+	// test case: When invoking the getCloudUser method with an empty user
+	// credential map, it must throw an InvalidParameterException.
+	@Test(expected = InvalidParameterException.class) // verify
+	public void testGetCloudUserWithEmptyUserCredentials() throws FogbowException {
+		// set up
+		Map<String, String> userCredentials = new HashMap<String, String>();
+
+		// exercise
+		this.plugin.getCloudUser(userCredentials);
+	}
+    
+	// test case: When invoking the getCloudUser method with a map of the user's
+	// credentials, without an access key, it must throw an
+	// InvalidParameterException.
+	@Test(expected = InvalidParameterException.class) // verify
+	public void testtestGetCloudUserWithoutAccessKey() throws FogbowException {
+		// set up
+		Map<String, String> userCredentials = new HashMap<String, String>();
+		userCredentials.put(AwsConstants.SECRET_ACCESS_KEY, ANY_VALUE);
+
+		// exercise
+		this.plugin.getCloudUser(userCredentials);
+	}
+	
+	// test case: When invoking the getCloudUser method with a map of the user's
+	// credentials, without a secret access key, it must throw an
+	// InvalidParameterException.
+	@Test(expected = InvalidParameterException.class) // verify
+	public void testGetCloudUserWithoutSecretAccessKey() throws FogbowException {
+		// set up
+		Map<String, String> userCredentials = new HashMap<String, String>();
+		userCredentials.put(AwsConstants.ACCESS_KEY, ANY_VALUE);
+
+		// exercise
+		this.plugin.getCloudUser(userCredentials);
+	}
+    
+	// test case: When invoking the getCloudUser method with a map of invalid user
+	// credentials, it must throw an UnauthenticatedUserException.
+	@Test(expected = UnauthenticatedUserException.class) // verify
+	public void testGetCloudUserWithInvalidUserCredenctials() throws FogbowException {
+		// set up
+		Map<String, String> userCredentials = new HashMap<String, String>();
+		userCredentials.put(AwsConstants.ACCESS_KEY, ANY_VALUE);
+		userCredentials.put(AwsConstants.SECRET_ACCESS_KEY, ANY_VALUE);
+
+		// exercise
+		this.plugin.getCloudUser(userCredentials);
+	}
+	
+	// test case: When invoking the getCloudUser method with a map of valid user
+	// credentials, this must return the cloud user.
+	@Test
```

Comment:
sometimes, mocking makes the code hard to read. I think we can also mock the authenticate method and test how the AwsV2IdentityProviderPlugin behaves depending on the return of the authentication (including the exception). 
This approach implies on ignoring the interaction with the AwsBasicCredentials, IamClient, StaticCredentialsProvider since the authentication and deadling with the authetication as a blackbox.
In fact, I believe this strategy of encapsulating the interaction with services in a mocked method (from the very class under test) will make our tests simpler to read and mantain. Obsviosly, it leads to some code not being covered by the unit tests (anyway, we can cover them by system tests, or even be the emulated cloud ones) @raonismaneoto @wassoares @ANKerD

Reply:
Very nice idea @thiagomanel . Makes a lot of sense and this approach can lead us to avoid many of the most confusing part of our tests: the mocking ones.

Reply:
I also agree... I will update the test using the new approach.